frontend/src/Context/WebSocketProvider.tsx

```diff
@@ -1,21 +1,54 @@
 import { useContext, useState, useEffect, createContext, type ReactNode } from "react";
 import { useAuthContext } from "./AuthProvider";
+import type { Notification } from "@/types/AppTypes";
+import { toast } from "sonner"; //function that calls the notif pop up
 
 type WebSocketContextType = {
   socket: WebSocket | null; //can be of type websocket or null ehrn there isnt a connection yet
   isConnected: boolean;
+  newNotification: Notification | null;
 };
 
 const WebSocketContext = createContext<WebSocketContextType>({
   socket: null,
   isConnected: false,
+  newNotification: null,
 });
 
 export function WebSocketProvider({ children }: { children: ReactNode }) {
   const [socket, setSocket] = useState<WebSocket | null>(null);
   const [isConnected, setIsConnected] = useState(false);
+  const [newNotification, setNewNotification] = useState<Notification | null>(null); //adding this state so we can broadcast the new notification being added to any component
   const { session } = useAuthContext(); //we dont want to establish socket connection if user isnt authenticated / logged in
 
+  function handleNewNotification(notification: Notification) {
+    if (notification.priority === "immediate") {
+      //if the notification is immediate priortiy we call toast function
```

Comment:
Based on our last discussion, you can use the toast as the "nudge feature" instead